core/src/main/java/com/valtech/aem/saas/core/http/response/TypeaheadDataExtractionStrategy.java

```diff
@@ -0,0 +1,60 @@
+package com.valtech.aem.saas.core.http.response;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.collections4.MapUtils;
+
+/**
+ * A strategy for extracting typeahead options.
+ */
+@RequiredArgsConstructor
+public final class TypeaheadDataExtractionStrategy implements SearchResponseDataExtractionStrategy<List<String>> {
+
+  private static final String TYPEAHEAD_SOLR_INDEX_FIELD_NAME_PREFIX = "autocomplete_text_";
```

Comment:
please remove `solr` out of the constant and getTypeaheadSolrIndexFieldName . We don't need to expose the technology